microsoft/PowerToys
# [KBM] Migrate Engine and Editor into separate processes

## Summary of the Pull Request

**What is this about:**
KBM remapping(aka KBM engine) and KBM editor are running inside of `PowerToys.exe` process. In this PR, we move out the Engine and the Editor into separate processes. Also, we changed the behaviour of KBM, now when either 'remap key' or 'remap shortcut' windows are opened we suspend remapping.
 
**What is include in the PR:** 

**How does someone test / validate:** 
- Run PowerToys
- Try to edit key/shortcut remapping. Verify that during editing remapping are suspended
- Verify that new remapping is applied
- Check `PowerToys.KeybordManager.exe` process priority. When running elevated priority should be real-time and when running as a user it should be high
- Try to restart PowerToys as administrator
- Disable/enable KBM
- Start only KBM engine. Test that remapping works.
- Verify that it is possible to start only one instance of KBM engine

## Quality Checklist


## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
